### ISSUE TITLE
Documentation for Next.js configuration seems incorrect

### ISSUE THREAD

**Issue by apennell** [styleguidist/react-styleguidist#1834]
### Problem

The [Next.js section](https://react-styleguidist.js.org/docs/webpack#nextjs) of the Styleguidist docs state:

> That's it, notice that we don't need to install webpack as it's already included by Next.js.

I tried following this in a Next.js app and only added the Babel presets, but Styleguidist wouldn't run and load the guide until I installed and configured Webpack. 

### Suggestion

This behavior was unexpected based on how I read the docs. Unless there's something I did wrong and I really shouldn't have needed to install and configure Webpack with my Next app, I'm happy to update the Next.js section of the Webpack config docs to reflect this. If there _is_ something else I missed, I'll gladly update the docs to include whatever that is!

### Additional details

Without installing `webpack`, running `npx styleguidist server` showed this error:
```
Webpack is required for Styleguidist, please add it to your project:

    npm install --save-dev webpack

See how to configure webpack for your style guide:
https://react-styleguidist.js.org/docs/webpack
```

After explicitly installing `webpack` but without adding a webpack config, `npx styleguide run` does "run" the project but can't parse the files and shows this warning:
```
Warning: No webpack config found. You may need to specify "webpackConfig" option in your style guide config:
https://react-styleguidist.js.org/docs/webpack
```

Once I added `webpackConfig` to `styleguide.config.js` (used the example copied in the "Custom webpack config" section), it ran fine.

**Comment by adamjw3** [comment]
I also have the same problem. Anyone got this working that can share the steps to get set up in next js

**Comment by quangvietnguyen** [comment]
same issue here, any fixes yet?

**Comment by mschl0ssCN** [comment]
Not sure if this will work for the `nextjs` folks, but i ran into the same problem with `create-react-app`, that is to say webpack is built into CRA but styleguidist wouldn't work with it until I added a custom `webpackConfig` key to the `module.exports` of `styleguide.config.js` to tell it how to handle various file types.  Essentially I had to redo this part of webpack.  Here's my config file, if it helps at least one person then I've done some good in the world :)...unless you're using styleguidist for evil :(

```js
webpackConfig: {
    // https://github.com/styleguidist/react-styleguidist/issues/1910
    plugins: [
      // Rewrites the absolute paths to those two files into relative paths
      new webpack.NormalModuleReplacementPlugin(
        /react-styleguidist\/lib\/loaders\/utils\/client\/requireInRuntime$/,
        'react-styleguidist/lib/loaders/utils/client/requireInRuntime'
      ),
      new webpack.NormalModuleReplacementPlugin(
        /react-styleguidist\/lib\/loaders\/utils\/client\/evalInContext$/,
        'react-styleguidist/lib/loaders/utils/client/evalInContext'
      ),
    ],
    module: {
      rules: [
        {
          test: /\.tsx?$/,
          use: {
            loader: 'ts-loader',
            options: {
              transpileOnly: true,
            },
          },
          exclude: /node_modules/,
        },
        {
          test: /\.(css)$/,
          use: 'css-loader',
        },
        {
          test: /\.(s(a|c)ss)$/,
          use: ['style-loader', 'css-loader', 'sass-loader'],
        },
        {
          test: [/\.(png|jpe?g|gif)$/i, /\.(ttf|eot)(\?[a-z0-9]+)?$/],
          use: 'file-loader',
        },
        {
          test: /\.svg$/,
          use: {
            loader: 'svg-url-loader',
            options: {
              limit: 10000,
            },
          },
        },
        {
          test: /\.woff(2)?(\?[a-z0-9]+)?$/,
          use: {
            loader: 'url-loader',
            options: {
              limit: 10000,
            },
          },
        },
      ],
    },
  },
```

**Comment by albjeremias** [comment]
nextjs now uses turbopack. not webpack, i guess that's the big issue. 
this is my `styleguide.config.js`

```
module.exports = {
    components: 'src/components/**/*.{jsx,tsx}',
    propsParser: require('react-docgen-typescript').withCustomConfig(
        './tsconfig.json'
    ).parse
};
```

didn't work, took a lot of time.. but says:
```

It usually happens when using third-party libraries, see possible solutions here:
https://react-styleguidist.js.org/docs/thirdparties

Warning: src/components/user/LoginForm/index.tsx matches a pattern defined in “components” or “sections” options in your style guide config but doesn’t export a component.

It usually happens when using third-party libraries, see possible solutions here:
https://react-styleguidist.js.org/docs/thirdparties

Module parse failed: Unexpected token (41:23)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|   title
| }) {
>   const selectedNetwork: Network = useRef(
|     store,
|     (state: GlobalState) => state.networks.selectedNetwork,

Learn how to add webpack loaders to your style guide:
https://react-styleguidist.js.org/docs/webpack
```

the configuration shared above gives me this error:
```
er
SyntaxError: Unexpected token ':'
/home/ajeremias/devel/helpbuttons.org/web/styleguide.config.js:14
    module: {
          ^

SyntaxError: Unexpected token ':'
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1033:15)
    at Module._compile (node:internal/modules/cjs/loader:1069:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at getConfig (/home/ajeremias/devel/helpbuttons.org/web/node_modules/react-styleguidist/lib/scripts/config.js:70:14)
    at Object.<anonymous> (/home/ajeremias/devel/helpbuttons.org/web/node_modules/react-styleguidist/lib/bin/styleguidist.js:56:12)
```

https://github.com/helpbuttons/helpbuttons/tree/dev/web

any kind of support would be greatful, and pull requests are very welcome.
